sdk/state/ingest.go

```diff
@@ -55,6 +55,11 @@ func (c *Channel) ingestTxToUpdateInitiatorEscrowAccountSequence(tx *txnbuild.Tr
 		return
 	}
 
+	// If the transaction is from an earlier sequence number, return.
+	if tx.SourceAccount().Sequence < c.initiatorEscrowAccount().SequenceNumber {
```

Comment:
💡 This probably makes little difference, however technically we should also skip updating is it is equal, because it would be a duplicate transaction.
```suggestion
	if tx.SourceAccount().Sequence <= c.initiatorEscrowAccount().SequenceNumber {
```